From 2ed870dafa4292fedf9b2988f06afd6e57e72594 Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Tue, 26 Sep 2017 11:42:38 +0200
Subject: [PATCH] fix memory leak when collecting performance data

We need to check the service_perfdata_fd/host_perfdata_fd as well when pushing
performance data onto our buffer. Checking the template does not work because
it is never NULL as it will be set to a default if its unset.

This should fix issue #200.
---
 src/naemon/perfdata.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/naemon/perfdata.c b/src/naemon/perfdata.c
index 5b90d517..8198e5ee 100644
--- a/src/naemon/perfdata.c
+++ b/src/naemon/perfdata.c
@@ -483,6 +483,9 @@ static int xpddefault_update_service_performance_data_file(nagios_macros *mac, s
 	if (svc == NULL)
 		return ERROR;
 
+	if (service_perfdata_fd < 0)
+		return OK;
+
 	if (service_perfdata_file_template == NULL)
 		return OK;
 
@@ -518,6 +521,9 @@ static int xpddefault_update_host_performance_data_file(nagios_macros *mac, host
 	if (hst == NULL)
 		return ERROR;
 
+	if (host_perfdata_fd < 0)
+		return OK;
+
 	if (host_perfdata_file_template == NULL)
 		return OK;
 
